Source/Spatializer/CrossPlatform/HrtfWrapper.cpp

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "HelperMacros.h"
+#include "HrtfWrapper.h"
+#include <exception>
+#include <cstring>
+
+// Statics
+std::unique_ptr<HrtfWrapper> HrtfWrapper::s_HrtfWrapper;
+
+HrtfWrapper::SourceInfo::SourceInfo(uint32_t index, HrtfInputBuffer* const sourceBuffer)
+    : m_SourceIndex(index), m_SourceBuffer(sourceBuffer)
+{
+}
+
+HrtfWrapper::SourceInfo::~SourceInfo()
+{
+    m_SourceBuffer->Buffer = nullptr;
+    m_SourceBuffer->Length = 0;
+    if (HrtfWrapper::s_HrtfWrapper)
+    {
+        s_HrtfWrapper->ReleaseSource(m_SourceIndex);
+    }
+}
+
+void HrtfWrapper::InitWrapper()
+{
+    RETURN_RESULT_IF_NULL(HrtfWrapper::s_HrtfWrapper);
+    HrtfWrapper::s_HrtfWrapper.reset(new HrtfWrapper());
+}
+
+bool HrtfWrapper::SourceInfo::SetParameters(HrtfAcousticParameters* params) const noexcept
+{
+    RETURN_RESULT_IF_NULL(HrtfWrapper::s_HrtfWrapper, false);
+    return HrtfWrapper::s_HrtfWrapper->SetParameters(m_SourceIndex, params);
+}
+
+float* HrtfWrapper::SourceInfo::GetBuffer() const noexcept
+{
+    return m_SourceBuffer->Buffer;
+}
+
+uint32_t HrtfWrapper::SourceInfo::GetIndex() const noexcept
+{
+    return m_SourceIndex;
+}
+
+HrtfWrapper::SourceInfo* HrtfWrapper::GetHrtfSource()
+{
+    RETURN_RESULT_IF_NULL(HrtfWrapper::s_HrtfWrapper, nullptr);
+    return HrtfWrapper::s_HrtfWrapper->GetAvailableHrtfSource();
+}
+
+uint32_t HrtfWrapper::Process(float* outputBuffer, uint32_t numSamples, uint32_t numChannels) noexcept
+{
+    RETURN_RESULT_IF_NULL(HrtfWrapper::s_HrtfWrapper, 0);
+    return HrtfWrapper::s_HrtfWrapper->ProcessHrtfs(outputBuffer, numSamples, numChannels);
+}
+
+HrtfWrapper::HrtfWrapper()
+    : m_GlobalReverbPower(0), m_GlobalReverbTime(1), m_SampleBuffers(c_HrtfMaxSources, c_HrtfFrameCount)
+{
+    for (auto&& buffer : m_HrtfInputBuffers)
+    {
+        buffer.Buffer = nullptr;
+        buffer.Length = 0;
+    }
+
+    THROW_IF_FALSE(
+        HrtfEngineInitialize(c_HrtfMaxSources, HrtfEngineType_Flex, c_HrtfFrameCount, &m_FlexEngine), std::bad_alloc);
+
+    THROW_IF_FALSE(
+        HrtfEngineSetOutputFormat(m_FlexEngine.Get(), HrtfOutputFormat_Stereo),
```

Comment:
Don't need this call. FLEX defaults to stereo (and I assume SetOutputFormat will go away soon)